### ISSUE TITLE
feature-request: option to disable remote commands

### ISSUE THREAD

**Issue by muetzekoeln** [E2OpenPlugins/e2openplugin-OpenWebif#970]
This is a feature request. I have some clients (smartphones, TV boxes) using my Dreambox for streaming live TV. But I do not want to allow them to send remote commands. Like to change channel or power-off. 

Please make an option to disable remote commands.

**Comment by jbleyel** [comment]
This is not so easy.
1) The definition of a to be disabled features list differs by the use case. 
2) Disable affects all clients and will break these features on all clients. (e.g. webinterface)

Here are some examples how to:
* use post requests
* use different user/password
* use session token
* ect....

**Comment by wedebe** [comment]
@muetzekoeln , from what I understand you just want to consume data (eg. epg) from the stb and essentially hide the remote control feature from the sidebar along with the header action buttons?

**Comment by muetzekoeln** [comment]
@wedebe: Yes, something like this. Just allow streaming live TV as well as
PVR, show EPG and maybe program timers. But not to change settings,
bouquets or header channel, or switching off/on.

Am Mi., 21. Okt. 2020 um 18:47 Uhr schrieb wedebe <notifications@github.com

**Comment by jbleyel** [comment]
Hi @muetzekoeln, das kann man schon machen, das würde jedoch den Code so extrem aufblasen und die Änderung würde sich auf alle auswirken.

Wenn überhaupt könnte man einen extra Branch machen wo diese Funktionen drin sind.

Allerdings verstehe ich immer noch nicht dein Problem.
Die Clients die du ansprichst sind doch bei dir oder nicht?

Einen Schalter "Disable Remote Commands" kann ich einbauen.
Dieser Schalter gilt aber dann für alle Clients. !!!!

**Comment by muetzekoeln** [comment]
Hallo,

ich will mal versuchen das zu erklären. Wir sind eine vierköpfige Familie
und nutzen eine Dreambox mit 2xDVB-S und 2xDVB-C. Es ist mir selber jetzt
schon mal passiert, dass ich mit dem PC über WebIf versehentlich meiner
Frau (vorm TV) den Kanal gewechselt haben, weil ich das falsche Icon
geklickt habe. Und ich möchte natürlich auch verhindern, dass meine Kindern
größeren Unfug anstellen können, z.B Einstellungen oder Boquetts ändern.
Die Hemmschwelle ist am PC oder Handy deutlich kleiner, als wenn man direkt
an der Box mit der Fernbedienung sitzt.

Im Moment ist das WebIf ja ein Zwitter. Einerseits eine Art Fernbedienung,
die einem die Arbeit mit der Infratot-Fernbedingung abnimmt. Andererseits
ein Remote-Streaming-Client. Für mich (und ich schätze mal für einige
andere Nutzer da draußen) ist nur die letztere Funktion interessant und die
andere führt bestenfalls zu Fehlbedienungen. Es wäre klasse, es gäbe eine
Einstellung bei der das WebIf nur auf Remote-Streaming-Funktionen reduziert
werden könnte.

Viele Grüße,
Christoph


Am Mi., 21. Okt. 2020 um 20:00 Uhr schrieb jbleyel <notifications@github.com

**Comment by jbleyel** [comment]
Alles klar.

Dann kann ich folgendes anbieten.
Es gibt X extra Settings für je eine Funktion die nicht gehen soll.
* No ZAP
* No RC
* No BQE
* ...

Diese Settings kann man dann in den Einstellungen im Openwebif AN/AUS schalten.
In den OSD Einstellungen würde ich das aber nicht machen.
Dies würde dann kaum Auswirkungen auf den Rest haben.
